src/coreclr/vm/ceeload.h

```diff
@@ -1384,7 +1381,12 @@ class Module : public ModuleBase
     }
 
     HRESULT GetScopeName(LPCUTF8 * pszName) { WRAPPER_NO_CONTRACT; return m_pPEAssembly->GetScopeName(pszName); }
-    const SString &GetPath() { WRAPPER_NO_CONTRACT; return m_pPEAssembly->GetPath(); }
+    const SString &GetPath()
+    {
+        WRAPPER_NO_CONTRACT;
+        _ASSERTE(SString{m_path}.Equals(m_pPEAssembly->GetPath()));
```

Comment:
I think you want this to be a little less format. I think you want to just check that the pointer values themselves are the same. In this case if `m_path` is stale (that is, freed), then this check could be correct, but not really. I would do something simpler:

```c
// Validate the pointers are the same to ensure the lifetime of m_path is handled.
_ASSERTE(m_path == m_pPEAssembly->GetPath().GetUnicode()));
```

Reply:
Agree that `GetUnicode()` does permute the value and avoiding it is a good thing, but that is actually what we want to know in this case. If it does permute the value "again" we should get another memory allocation and the pointers wouldn't be equal. Otherwise, it is what it was before. Either way, the check accomplishes the goal.